packages/fast-usdc/src/cli/lp-commands.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @import {Command} from 'commander';
+ * @import {OfferSpec} from '@agoric/smart-wallet/src/offers.js';
+ * @import {ExecuteOfferAction} from '@agoric/smart-wallet/src/smartWallet.js';
+ */
+
+import { fetchEnvNetworkConfig, makeVstorageKit } from '@agoric/client-utils';
+import { Nat } from '@endo/nat';
+import { InvalidArgumentError } from 'commander';
+import { outputActionAndHint } from './bridge-action.js';
+
+/** @param {string} arg */
+const parseNat = arg => {
+  try {
+    const n = Nat(BigInt(arg));
+    return n;
+  } catch {
+    throw new InvalidArgumentError('Not a number');
+  }
+};
+
+/**
+ * @param {Command} program
+ * @param {{
+ *   fetch?: Window['fetch'];
+ *   vstorageKit?: Awaited<ReturnType<typeof makeVstorageKit>>;
+ *   stdout: typeof process.stdout;
+ *   stderr: typeof process.stderr;
+ *   env: typeof process.env;
+ *   now: typeof Date.now;
+ * }} io
+ */
+export const addLPCommands = (
+  program,
+  { fetch, vstorageKit, stderr, stdout, env, now },
+) => {
+  const operator = program
+    .command('lp')
+    .description('Liquidity Provider commands');
+
+  const loadVsk = async () => {
+    if (vstorageKit) {
+      return vstorageKit;
+    }
+    assert(fetch);
+    const networkConfig = await fetchEnvNetworkConfig({ env, fetch });
+    return makeVstorageKit({ fetch }, networkConfig);
+  };
+  /** @type {undefined | ReturnType<typeof loadVsk>} */
+  let vskP;
+
+  operator
+    .command('deposit')
+    .description('Deposit USDC into pool')
+    .addHelpText(
+      'after',
+      '\nPipe the STDOUT to a file such as deposit.json, then use the Agoric CLI to broadcast it:\n  agoric wallet send --offer accept.json --from gov1 --keyring-backend="test"',
+    )
+    .requiredOption('--amount <number>', 'uUSDC amount', parseNat)
```

Comment:
Let's not make people write `1000000` for 1 USDC. **important**

Use `parseDecimal` here (based on `assertParsableNumber`) and then once you have the `usdc` brand from vstorage, `parseRatio` is really handy:

```js
const USDC_DECIMALS = 6;

/** @param {string} arg */
  const unit = AmountMath.make(usdc, 10n ** BigInt(USDC_DECIMALS));
  const usdcAmount = multiplyBy(unit, parseRatio(arg, usdc));
}
```

Reply:
Done